jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannelState.java

```diff
@@ -1215,14 +1215,15 @@ public boolean onReadEof()
         try(Locker.Lock lock= _locker.lock())
         {
             if (LOG.isDebugEnabled())
-                LOG.debug("onReadEof {}",toStringLocked());
+                LOG.debug("onEof {}",toStringLocked());
 
+            // Force read interest so onAllDataRead can be called
+            _asyncRead=Interest.REGISTERED;
```

Comment:
This is the fix for #1317 as reading -1 will force read interest to be registered, so the normal tests in unhandle will work